modules/tinymce/src/core/test/ts/browser/ReadOnlyModeTest.ts

```diff
@@ -223,6 +234,33 @@ UnitTest.asynctest('browser.tinymce.core.ReadOnlyModeTest', (success, failure) =
         UiFinder.sWaitFor('Waited for menu', SugarBody.body(), '.tox-menu'),
         sSetMode('readonly'),
         UiFinder.sNotExists(SugarBody.body(), '.tox-menu')
+      ]),
+      Log.stepsAsStep('TINY-6248', 'getAnchorHrefOpt should return an Optional of the href of the closest anchor tag', [
+        tinyApis.sSetContent('<p><a href="https://tiny.cloud">external link</a></p>'),
+        sAssertHrefOpt('a', Optional.some('https://tiny.cloud')),
+        tinyApis.sSetContent('<p><a>external link with no href</a></p>'),
+        sAssertHrefOpt('a', Optional.none()),
+        tinyApis.sSetContent('<p><a href="https://tiny.cloud"><img src="">nested image </img>inside anchor</a></p>'),
+        sAssertHrefOpt('img', Optional.some('https://tiny.cloud'))
+      ]),
+      Log.stepsAsStep('TINY-6248', 'processReadonlyEvents should scroll to bookmarks', [
```

Comment:
It may not be necessary but would it be worth doing a test with `<a name="someBookmark"></a>` as well?

Reply:
Yeah, I think I was so keen to just have the `NamedChain` complete that I forgot that case. Will fix